neural_compressor/torch/algorithms/weight_only/save_load.py

```diff
@@ -37,14 +39,596 @@ def save(model, output_dir="./saved_results"):
 
 
-def load(output_dir="./saved_results"):
-    qmodel_file_path = os.path.join(os.path.abspath(os.path.expanduser(output_dir)), WEIGHT_NAME)
-    model = torch.load(qmodel_file_path)
-    logger.info("Quantized model loading successful.")
+def load(model_name_or_path, original_model=None, format=LoadFormat.DEFAULT, device="cpu", **kwargs):
+    """Load quantized weight-only quantization model.
+
+    1. Load INC weight-only quantized model in local.
+    2. Load HuggingFace weight-only quantized model,
+       including GPTQ/AWQ models and upstreamed INC quantized models in HF model hub.
+
+    Args:
+        model_name_or_path (str):  torch checkpoint directory or hugginface model_name_or_path.
+            If 'format' is set to 'huggingface', it means the huggingface model_name_or_path.
+            If 'format' is set to 'default', it means the 'checkpoint_dir'.
+            Parameter should not be None. it coworks with 'original_model' parameter to load INC
+            weight-only quantized model in local.
+        original_model (torch.nn.module, optional): original model before quantization.
+            Needed if 'format' is set to 'default' and not TorchScript model.Defaults to None.
+        format (str, optional): 'defult' for loading INC weight-only quantized model.
+            'huggingface' for loading huggingface WOQ causal language model. Defaults to "default".
+        kwargs (remaining dictionary of keyword arguments, optional):
+            remaining dictionary of keyword arguments for loading huggingface models.
+            will be passed to the huggingface model's `__init__` method, such as 'trust_remote_code', 'revision'.
+    Returns:
+        torch.nn.Module: quantized model
+    """
+    model_loader = WOQModelLoader(model_name_or_path, original_model, format, device="cpu", **kwargs)
+    model = model_loader.load_woq_model()
     return model
+
+
+class WOQModelLoader:
+    def __init__(self, model_name_or_path, original_model=None, format=LoadFormat.DEFAULT, device="cpu", **kwargs):
+        # TODO: When loading WOQ model, use different WeightOnlyLinear module according to device.
+        self.model_name_or_path = model_name_or_path
+        self.original_model = original_model
+        self.format = format
+        self.device = device
+        self.kwargs = kwargs
+        self.quantization_config = {}
+        self.loaded_state_dict_keys = {}
+
+    def load_woq_model(self):
+        if self.format == LoadFormat.HUGGINGFACE:
+            model = self.load_hf_format_woq_model()
+            logger.info("Quantized huggingface model loading successful.")
+        elif self.format == LoadFormat.DEFAULT:
+            qmodel_weight_file_path = os.path.join(
+                os.path.abspath(os.path.expanduser(self.model_name_or_path)), WEIGHT_NAME
+            )
+            assert os.path.exists(qmodel_weight_file_path), "Cannot load model weight from path {}".format(
```

Comment:
It only checks the existence of the file but does not make sure it's a qmodel weight.
Suggest raising a message to guide users on which values should be passed if they want to load the qmodel with `DEFAULT` format.

Reply:
add log info in https://github.com/intel/neural-compressor/commit/a33c766e8865f009ab02fa4ccd24cedd85574113